tools/test/test_gen_backend_stubs.py

```diff
@@ -238,7 +238,7 @@ def test_unrecognized_key(self) -> None:
         output_error = self.get_errors_from_gen_backend_stubs(yaml_str)
         self.assertExpectedInline(
             output_error,
-            """ contains unexpected keys: invalid_key. Only the following keys are supported: backend, class_name, cpp_namespace, extra_headers, supported, autograd, full_codegen, non_native""",  # noqa: B950
+            """ contains unexpected keys: invalid_key. Only the following keys are supported: backend, class_name, cpp_namespace, extra_headers, supported, autograd, full_codegen, non_native, ir_gen""",  # noqa: B950
```

Comment:
All of these extra flags are technically leaking into the `gen_backend_stubs.py` codegen file... so maybe we should think about a way to decouple them some more.

Not gonna block this PR on that though 😛